src/app/view/components/login/login-page-resolver.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Resolve } from '@angular/router';
+import { Injectable } from '@angular/core';
+import { LoginService } from 'dotcms-js';
+import { Observable } from 'rxjs';
+import { DotLoginInformation, DotSystemInformation } from '@models/dot-login';
+import { pluck, tap } from 'rxjs/operators';
+
+export const LOGIN_LABELS = [
+    'email-address',
+    'user-id',
+    'password',
+    'remember-me',
+    'sign-in',
+    'get-new-password',
+    'cancel',
+    'Server',
+    'error.form.mandatory',
+    'angular.login.component.community.licence.message',
+    'reset-password-success',
+    'a-new-password-has-been-sent-to-x',
+    'welcome-back',
+    'forgot-password',
+    'get-new-password',
+    'cancel',
+    'an-email-with-instructions-will-be-sent',
+    'reset-password',
+    'enter-password',
+    're-enter-password',
+    'change-password',
+    'reset-password-confirmation-do-not-match'
+];
+
+/**
+ *
+ * @export
+ * @class LoginPageResolver
+ * @implements {Resolve<any>}
+ */
+@Injectable()
+export class LoginPageResolver implements Resolve<DotLoginInformation> {
+    constructor(private loginService: LoginService) {}
+
+    resolve(): Observable<DotLoginInformation> {
+        return this.loginService.getLoginFormInfo('', LOGIN_LABELS).pipe(
+            pluck('bodyJsonObject'),
+            tap((loginInfo: DotLoginInformation) => {
+                document.body.style.backgroundColor = this.setBackgroundColor(loginInfo.entity);
```

Comment:
The resolver only care about get data not what to do with it... setting the body style should be done in the component.